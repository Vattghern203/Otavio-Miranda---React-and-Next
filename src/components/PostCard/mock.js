export const postCardPropsMock = {
    id: 1,
    title: "placeholder",
    body: "random text",
    cover: "img/cool-img.jpg"
}