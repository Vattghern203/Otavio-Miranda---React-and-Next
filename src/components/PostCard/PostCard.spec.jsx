import { render, screen } from "@testing-library/react"
import { PostCard } from "."
import { postCardPropsMock } from "./mock"

const props = postCardPropsMock

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
        render(<PostCard {...props} />)

        expect(screen.getByRole('img')).toHaveAttribute('src', props.cover)
        
        expect(screen.getByRole('heading', {name: 'placeholder'})).toBeInTheDocument()

        expect(screen.getByText('random text')).toBeInTheDocument()
    })

    it('should match the snapshot', () => {
        const { container } = render(<PostCard {...props} />)

        expect(container.firstChild).toMatchSnapshot()
    })
})