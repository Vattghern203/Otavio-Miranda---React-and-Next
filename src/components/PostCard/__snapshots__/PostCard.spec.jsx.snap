// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match the snapshot 1`] = `
<div
  class="post"
  title="placeholder"
>
  <img
    alt="placeholder"
    src="img/cool-img.jpg"
  />
  <div
    class="post-content"
  >
    <h2>
      placeholder
    </h2>
    <p>
      random text
    </p>
  </div>
</div>
`;
